### PR TITLE
[bugfix] dyups message loss after worker coredump

### PR DESCRIPTION
**Pull request by FengXingYuXin**
1 if your _dyups_upstream_conf is not open and worker coredump, the new worker's dyups messages(from the time of master process's last reload oper) will loss。

### CODE REVIEW

**Comment by wangfakang** [comment]
@FengXingYuXin  Could you add some test cases for the scenario ? 
Thank you very much.

File: modules/ngx_http_upstream_dyups_module/ngx_http_dyups_module.c [hunk 1]
```diff
@@ -2125,6 +2148,141 @@ ngx_http_dyups_read_msg_locked(ngx_event_t *ev)
 }
 
 
+static void ngx_dyups_restore_upstreams_from_hist_queue()
```
Comment (chobits) [inline modules/ngx_http_upstream_dyups_module/ngx_http_dyups_module.c]: hist means `history`?

File: modules/ngx_http_upstream_dyups_module/ngx_http_dyups_module.c [hunk 2]
```diff
@@ -745,6 +756,13 @@ ngx_dyups_delete_upstream(ngx_str_t *name, ngx_str_t *rv)
         status = NGX_HTTP_INTERNAL_SERVER_ERROR;
```
Comment (chobits) [inline modules/ngx_http_upstream_dyups_module/ngx_http_dyups_module.c]: should added `goto finish`

Comment (FengXingYuXin) [reply modules/ngx_http_upstream_dyups_module/ngx_http_dyups_module.c]: thanks for your remind, and I have added it.

File: modules/ngx_http_upstream_dyups_module/ngx_http_dyups_module.c (hunk 1 above)
Comment (FengXingYuXin) [reply modules/ngx_http_upstream_dyups_module/ngx_http_dyups_module.c]: yes, if it's improper for tengine, i can adjust it.

**Comment by FengXingYuXin** [comment]
> @FengXingYuXin Could you add some test cases for the scenario ?
> Thank you very much.

OK, thanks for your remind, I will supply it later.

**Comment by chobits** [comment]
hi all

Because https://github.com/alibaba/tengine/pull/1255 (updated to yzprofile/dyups) has been merged.

This pr has conflicts again.

@FengXingYuXin , you should open a pr to yzprofile/dyups first,  our dev team will review your pr in yzprofile/dyups and then merge it to tengine/dyups.

**Comment by FengXingYuXin** [comment]
> hi all
> 
> Because #1255 (updated to yzprofile/dyups) has been merged.
> 
> This pr has conflicts again.
> 
> @FengXingYuXin , you should file a pr to yzprofile/dyups first, our dev team will review your pr in yzprofile/dyups and then merge it to tengine/dyups.

thanks I have create a pr：https://github.com/yzprofile/ngx_http_dyups_module/pull/124

**Pull request closed without merging**